From 44d17eff54cc7406f86c025bfbb38c3024dd8614 Mon Sep 17 00:00:00 2001
From: kk <405437445@qq.com>
Date: Mon, 23 Mar 2026 10:00:26 +0800
Subject: [PATCH 4/6] Add ZaparooEnabled setting to optionally disable Zaparoo
 service check

- Add ZaparooEnabled setting (default: false) to es_settings.cfg
- Skip Zaparoo service check at startup when disabled
- Reduce connection timeout from 3000ms to 500ms for faster response
- Users can enable by setting ZaparooEnabled=true in es_settings.cfg
---
 es-app/src/ZaparooSupport.cpp | 2 +-
 es-app/src/main.cpp           | 5 ++++-
 es-core/src/Settings.cpp      | 2 ++
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/es-app/src/ZaparooSupport.cpp b/es-app/src/ZaparooSupport.cpp
index 1fbc618..2a231df 100644
--- a/es-app/src/ZaparooSupport.cpp
+++ b/es-app/src/ZaparooSupport.cpp
@@ -21,7 +21,7 @@ void Zaparoo::checkZaparooEnabledAsync(const std::function<void(bool enabled)>&
 {
 	std::thread([func]()
 	{
-		bool enabled = isZaparooEnabled(3000);
+		bool enabled = isZaparooEnabled(500);
 		if (func)
 			func(enabled);
 	}).detach();
diff --git a/es-app/src/main.cpp b/es-app/src/main.cpp
index 7eb8e44..953fae1 100644
--- a/es-app/src/main.cpp
+++ b/es-app/src/main.cpp
@@ -532,7 +532,10 @@ int main(int argc, char* argv[])
 	HttpReq::resetCookies();
 	Genres::init();
 	MetaDataList::initMetadata();
-	Zaparoo::checkZaparooEnabledAsync();
+
+	// Check Zaparoo service only if enabled in settings
+	if (Settings::getInstance()->getBool("ZaparooEnabled"))
+		Zaparoo::checkZaparooEnabledAsync();
 
 	Window window;
 	SystemScreenSaver screensaver(&window);
diff --git a/es-core/src/Settings.cpp b/es-core/src/Settings.cpp
index 4d00003..4ddb329 100644
--- a/es-core/src/Settings.cpp
+++ b/es-core/src/Settings.cpp
@@ -175,6 +175,8 @@ void Settings::setDefaults()
 	mStringMap["HiddenSystems"] = "";
 	mStringMap["SystemConfigMode"] = "full";
 
+	mBoolMap["ZaparooEnabled"] = false;
+
 	mBoolMap["PublicWebAccess"] = false;	
 	mBoolMap["FirstJoystickOnly"] = false;
     mBoolMap["EnableSounds"] = false;
-- 
2.43.0

